src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/inbound/NhsAckResponseTest.java

```diff
@@ -0,0 +1,192 @@
+ * There is an NHSACK status code for each type of response:
+ *
+ * IAF: Valid Interchange, all Messages accepted
+ * IAP: Valid Interchange, some Messages accepted, some Messages rejected
+ * IRA: Valid Interchange, but all Messages were rejected
+ * IRM: Interchange rejected due to an error in one message
+ * IRI: Interchange rejected due to interchange level error
+ */
+@DirtiesContext
+class NhsAckResponseTest extends IntegrationBaseTest {
+    @Value("classpath:edifact/pathology_IAF.edifact.dat")
+    private Resource edifactIAFResource;
+
+    @Value("classpath:edifact/pathology_IAF_regex.nhsack.dat")
+    private Resource nhsAckIAFResource;
+
+    @Value("classpath:edifact/pathology_IAP.edifact.dat")
+    private Resource edifactIAPResource;
+
+    @Value("classpath:edifact/pathology_IAP_regex.nhsack.dat")
+    private Resource nhsAckIAPResource;
+
+    @Value("classpath:edifact/pathology_IRA.edifact.dat")
+    private Resource edifactIRAResource;
+
+    @Value("classpath:edifact/pathology_IRA_regex.nhsack.dat")
+    private Resource nhsAckIRAResource;
+
+    @Value("classpath:edifact/pathology_IRM.edifact.dat")
+    private Resource edifactIRMResource;
+
+    @Value("classpath:edifact/pathology_IRM_regex.nhsack.dat")
+    private Resource nhsAckIRMResource;
+
+    @Value("classpath:edifact/pathology_IRI.edifact.dat")
+    private Resource edifactIRIResource;
+
+    @Value("classpath:edifact/pathology_IRI_regex.nhsack.dat")
+    private Resource nhsAckIRIResource;
+
+    @MockBean(name = "timestampService")
+    private TimestampService timestampService;
+
+    @MockBean(name = "sequenceService")
+    private SequenceService sequenceService;
+
+    private static final Instant FIXED_TIME = Instant.parse("2020-04-27T16:37:00Z");
+    private static final Long FIXED_SEQUENCE_NUMBER = 1000L;
+
+    @BeforeEach
+    void setUp() {
+        when(timestampService.getCurrentTimestamp()).thenReturn(FIXED_TIME);
+        when(sequenceService.generateInterchangeSequence(any(), any())).thenReturn(FIXED_SEQUENCE_NUMBER);
+        clearGpOutboundQueue();
+        clearMeshMailboxes();
+        clearMeshOutboundQueue();
+    }
+
+    @Test
+    void whenValidEdifactSentThenCorrectNhsAckReturned()
+            throws IOException {
+
+        final String content = new String(Files.readAllBytes(edifactIAFResource.getFile().toPath()));
+
+        final MeshMessage meshMessage = new MeshMessage()
+                .setWorkflowId(WorkflowId.PATHOLOGY)
+                .setContent(content);
+
+        sendToMeshInboundQueue(meshMessage);
+
+        final var nhsAck = waitForMeshMessage(getMeshClient());
+
+        assertThat(nhsAck.getWorkflowId()).isEqualTo(WorkflowId.PATHOLOGY_ACK);
+
+        final String nhsAckContent = nhsAck.getContent();
+        final String expectedContent = new String(Files.readAllBytes(nhsAckIAFResource.getFile().toPath())).
+                replace("\n", "");
+        assertThat(nhsAckContent).matches(expectedContent);
```

Comment:
I think this is a good approach, but it might be cleaner with mustache placeholders in the `*regex.nhsack.dat` files.
```
UNB+UNOC:3+TES5+XX11+{{time}}++PATH_MEDRPT_V3_ACK'
```

Maybe something for us to think about later though.

Reply:
After discussion, it was decided to leave this for now.